### DISCUSSION TITLE
Infra 구성 어떻게 하면 좋을까요?

### DISCUSSION THREAD

**Discussion by Dr-KoKo** [Strong-Potato/TripVote-BE#12] [General]
간단하게 구성도 그려보았는데 이외에 고려사항이 더 있으면 말씀해주시면 감사하겠습니다.
![infra drawio](https://github.com/Strong-Potato/TripVote-BE/assets/97681286/f67abbda-c5bc-48d9-a224-4d27f5b28b17)

**Comment by laigasus** [comment]
Place API 가공처리를 어떻게 할지 고민입니다...
1. Tripvote 앱 내에 멀티모듈로 사용하여 처리
2. 별도 서버 운영하여 메인(Tripvote) 앱 과 통신

**Reply by laigasus** [reply]
멀티 모듈로 하면, 프로젝트 구성이 이리 되려나요

1. external. OpenAPI(TourAPI, 구글 PlaceAPI 가공)
2. notification. 알림 redis pub/sub
3. api. 도메인 비즈니스 코드 수행(mysql DB 통신)

**Reply by Dr-KoKo** [reply]
별도 서버에서 Place API 가공처리를 한다면 EC2 인스턴스를 하나 더 만들어야겠네요.
또 알림 서버도 하나 필요할 것 같아서 수정 후 반영하겠습니다.

**Reply by laigasus** [reply]
멀티모듈로 했을때, 인스턴스 따로 생성하지 않아도 된다면 프로젝트 구성을 바꿔야 할 것 같아요.
해보고 될 것 같으면 공유하겠습니다
